File: Module3/BaiTap/ProductManager/src/main/java/service/ProductService.java
package service;

import dao.ProductDAO;
import model.Product;

import java.util.List;

public class ProductService {
    private ProductDAO productDAO = new ProductDAO();

    public List<Product> getAll() {
        return  productDAO.getAll();
    }
    public Product selectProductByID(int id) {
        return productDAO.selectProductByID(id);
    }

    public List<Product> findByName(String name) {
        return productDAO.findByName(name);
    }

    public void add(Product product) {
        productDAO.add(product);
    }

    public void update(Product product) {
        productDAO.update(product);
    }

    public void delete(int id) {
        productDAO.delete(id);
    }
}

File: Module5/ss9_Connect_BE/angular-product-management/src/app/service/product.service.ts
import { Injectable } from '@angular/core';
import {Product} from '../model/product';
import {HttpClient} from '@angular/common/http';
import {environment} from '../../environments/environment';
import {Observable, Subject} from 'rxjs';
import {tap} from 'rxjs/operators';
const API_URL = `${environment.apiUrl}`;
@Injectable({
  providedIn: 'root'
})
export class ProductService {

  private _refreshNeeded = new Subject<void>();

  get refreshNeeded() {
    return this._refreshNeeded;
  }

  constructor(private http: HttpClient) { }

  getAll(): Observable<Product[]> {
    return this.http.get<Product[]>(API_URL + '/products');
  }

  save(product: Product): Observable<Product> {
    return this.http.post<Product>(API_URL + '/products', product).pipe(
      tap(() => {
        this._refreshNeeded.next();
      })
    );
  }

  update(product: Product): Observable<Product> {
    return this.http.put<Product>(`${API_URL}/products/${product.id}`, product);
  }

  findById(id: number): Observable<Product> {
    return this.http.get<Product>(`${API_URL}/products/${id}`);
  }

  delete(id: number): Observable<Product> {
    return this.http.delete<Product>(`${API_URL}/products/${id}`);
  }
}
